src/Api/PubnubApi/EventEngine/Subscribe/Invocations/SubscriptionInvocations.cs

```diff
@@ -3,17 +3,38 @@
 
 namespace PubnubApi.PubnubEventEngine.Subscribe.Invocations {
 	internal class EmitMessagesInvocation : Core.IEffectInvocation {
-		public List<object> messages;
+		public IEnumerable<PNMessageResult<object>> Messages;
+
+		public EmitMessagesInvocation(IEnumerable<PNMessageResult<object>> messages)
+		{
+			this.Messages = messages;
+		}
 	}
 
 	internal class EmitStatusInvocation : Core.IEffectInvocation {
-		public IEnumerable<string> Channels;
-		public IEnumerable<string> ChannelGroups;
+		// TODO merge status variables into one?
+		public PNStatusCategory StatusCategory;
```

Comment:
Do we need separate PNStatusCategory when we already have Category in PNStatus?

Reply:
This is still used by disconnect/reconnect, will see if we can further simplify